pkg/routeagent_driver/handlers/ovn/connection.go

```diff
@@ -186,45 +188,85 @@ func getTLSConfig(k8sClientset clientset.Interface) (*tls.Config, error) {
 	return tlsConfig, nil
 }
 
-func discoverOvnKubernetesNetwork(ctx context.Context, k8sClientset clientset.Interface, zoneName string) (string, error) {
-	ovnDBPod, err := FindPod(ctx, k8sClientset, "name=ovnkube-db")
+func discoverOvnKubernetesNetwork(ctx context.Context, k8sClientset clientset.Interface,
+	dynamicClient dynamic.Interface, zoneName string,
+) (string, error) {
+	openshiftNetwork, err := FindOpenshiftNetwork(ctx, dynamicClient)
 	if err != nil {
 		return "", err
 	}
 
-	var nbdbAddress string
+	if openshiftNetwork != nil {
+		return discoverOpenshiftOvnKubernetesNetwork(ctx, k8sClientset)
+	}
+
+	return discoverKindOvnKubernetesNetwork(ctx, k8sClientset, zoneName)
+}
+
+/*
+
+openshift non-ic: in this case there will get the "app=ovnkube-node" and check for ovnkube-db service,
+if present we will the default Openshift db path
+
+openshift ic with one node per zone: if above service is not present we will return default Openshift socket path
+*/
+
+func discoverOpenshiftOvnKubernetesNetwork(ctx context.Context, k8sClientSet clientset.Interface) (string, error) {
+	ovnPod, err := FindPod(ctx, k8sClientSet, "app=ovnkube-node")
```

Comment:
At a couple of places, we are using "app=ovnkube-node". This can be replaced with a constant.

Reply:
Done